src/app/InteractionModelEngine.cpp

```diff
@@ -517,20 +517,7 @@ CHIP_ERROR InteractionModelEngine::ParseAttributePaths(const Access::SubjectDesc
             ConcreteAttributePath concretePath(paramsList.mValue.mEndpointId, paramsList.mValue.mClusterId,
                                                paramsList.mValue.mAttributeId);
 
-// "check" method to ensure DM interface and ember interface are identical
-#if CHIP_CONFIG_USE_DATA_MODEL_INTERFACE && CHIP_CONFIG_USE_EMBER_DATA_MODEL
-            VerifyOrDie(GetDataModelProvider()->GetAttributeInfo(concretePath).has_value() ==
-                        ConcreteAttributePathExists(concretePath));
-#endif
-
-            if (
-#if CHIP_CONFIG_USE_DATA_MODEL_INTERFACE
-                GetDataModelProvider()->GetAttributeInfo(concretePath).has_value()
-
-#else
-                ConcreteAttributePathExists(concretePath)
-#endif
-            )
+            if (IsExistingAttributePath(concretePath))
```

Comment:
Is this supposed to be checking for this being an "Existent Path" in the spec terminology?  If so, the naming should match the spec, no?  IsExistentAttributePath.

It looks like this function was added in https://github.com/project-chip/connectedhomeip/pull/35746 after I had reviewed it....  I was being confused why I wasn't recognizing it.

Reply:
I will rename. It was added after reviewer request (Tennessee) of not having ifdefs in the middle of an if condition because that is ugly to read ... I agreed.